target/linux/ramips/dts/mt7621_oraybox_p5.dts

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "oraybox,p5", "mediatek,mt7621-soc";
+	model = "OrayBox P5";
+
+	aliases {
+		led-boot = &led_status_green;
+		led-failsafe = &led_status_red;
+		led-running = &led_status_blue;
+		led-upgrade = &led_status_green;
+		label-mac-device = &gmac0;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status_blue: status-blue {
+			label = "blue:status";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status_red: status-red {
+			label = "red:status";
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status_green: status-green {
+			label = "green:status";
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	reg_usb_vbus: regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "usb_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		#gpio = <&gpio 12 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x80000>;
+			read-only;
+		};
+
+		partition@80000 {
+			label = "kpanic";
+			reg = <0x80000 0x20000>;
+		};
+
+		factory: partition@100000 {
+			label = "factory";
+			reg = <0x100000 0x40000>;
+			read-only;
+		};
+
+		firmware: partition@140000 {
+			label = "firmware";
+			reg = <0x140000 0x2000000>;
+
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			kernel: partition@0 {
+				label = "kernel";
+				reg = <0x0 0x400000>;
+			};
+			
+			ubi: partition@400000 {
+				label = "ubi";
+				reg = <0x400000 0x1c00000>;
+			};
+		};
+		
+		partition@2140000 {
+			label = "debug";
+			reg = <0x2140000 0x80000>;
+		};
+
+		partition@21c0000 {
+			label = "bdinfo";
+			reg = <0x21c0000 0x80000>;
+			read-only;
+
+			compatible = "nvmem-cells";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			macaddr_bdinfo_9: macaddr@9 {
+				reg = <0x9 0x6>;
+			};
+		};
+
+		partition@2240000 {
+			label = "reserve";
+			reg = <0x2240000 0x80000>;
+			read-only;
+		};
+
+		partition@22c0000 {
+			label = "overlay";
+			reg = <0x22c0000 0x5cc0000>;
+		};
+
+		partition@7f80000 {
+			label = "opt";
+			reg = <0x7f80000 0x2cc0000>;
```

Comment:
Something is wrong with the partition table. It spanning over 180MB, but the flash chip only has 128MB.
Either the flash chip is bigger or the partition layout is wrong.

Reply:
@hauke Modified and tested successfully